builtin/providers/azurerm/resource_arm_virtual_machine.go

```diff
@@ -553,11 +553,13 @@ func resourceArmVirtualMachineCreate(d *schema.ResourceData, meta interface{}) e
 		}
 	}
 
-	osProfile, err := expandAzureRmVirtualMachineOsProfile(d)
-	if err != nil {
-		return err
+	if _, ok := d.GetOk("os_profile"); ok {
+		osProfile, err := expandAzureRmVirtualMachineOsProfile(d)
+		if err != nil {
+			return err
+		}
+		properties.OsProfile = osProfile
 	}
```

Comment:
Ok @tombuildsstuff I beleive I got the changes in.  Of course I had to add more code.  This is why we test.  Thanks for your guidance.

I ran Standard_F2 which shaved about 2 minutes from the Standard_D1_v2.